File: server/dotnet/sdk-dotnet-example/Business/Domain/Common/AddressDto.cs
using Business.Domain.Common.Enums;

namespace Business.Domain.Common;

public class AddressDto
{
    public string? FirstName { get; set; }
    
    public string? LastName { get; set; }
    
    public Country? Country { get; set; }
    
    public string? Zip { get; set; }
    
    public string? City { get; set; }
    
    public string? Street { get; set; }
}

File: server/nodejs/src/business/dtos/common/address-dto.ts
import { Country } from '../../domain/common/enums/country';

export type AddressDto = {
    firstName?: string | null;
    lastName?: string | null;
    country?: Country | null;
    zip?: string | null;
    city?: string | null;
    street?: string | null;
};

export function fromJson(data: any): AddressDto {
    return {
        firstName: data?.firstName,
        lastName: data?.lastName,
        country: data?.country ? (data.country as Country) : undefined,
        zip: data?.zip,
        city: data?.city,
        street: data?.street,
    };
}

export function toJson(dto: AddressDto): Record<string, any> {
    return {
        firstName: dto.firstName,
        lastName: dto.lastName,
        country: dto.country,
        zip: dto.zip,
        city: dto.city,
        street: dto.street,
    };
}
